codestodians/food-basket-api
# [Feature: #154589791] Deactivate a user's account.

#### What does this PR do?
- Enables User account deactivation.

#### Description of Task to be completed?
- Create mutation for deactivating a user account.
- Write test cases for different account deactivation scenarios.
- Update Payment test suite (to catch uncovered lines).
- Prevent Login for deactivated user.
- Prevent details update for deactivated user.

#### How should this be manually tested?
- Check out into the `ft-deactivate-user-154589791` branch.
- Run the `deactivateAccount` mutation on `graphiql` for: A. A fresh deactivation B. An already deactivated account. C. A non-existent account
- Ensure that you get the correct messages/results every time.
- Run tests with `coverage run --source=food_basket --omit */snapshots/* manage.py test`
- Ensure all lines are covered with `coverage report -m`

#### Any background context you want to provide?
- When a user no longer has the need for their account, there should be an option to deactivate said account (not delete, so that records can be kept, and the user has the option to reactivate the account in the future).

#### What are the relevant pivotal tracker stories?
- [#154589791](https://www.pivotaltracker.com/story/show/154589791)
